crates/noirc_evaluator/src/ssa/anchor.rs

```diff
@@ -175,6 +191,12 @@ impl Anchor {
                 MemItem::Const(p) | MemItem::ConstLoad(p) => {
                     let a = self.get_mem_map(array_id);
                     let b_idx = b_value.to_u128() as usize;
+                    if b_idx >= a.len() {
+                        return Some(CseAction::Error(RuntimeErrorKind::ArrayOutOfBounds {
```

Comment:
The function return type should be `Result<Option<CseAction>, RuntimeErrorKind>` since it doesn't matter that the RuntimeError came from a CseAction.

Your code would then roughly be:

```rust
return Err(RuntimeErrorKind::ArrayOutOfBounds{ index : b_idx as u128, bound : a.len() as u128})
```

Reply:
Done.

Note that IMHO handling both an Option and a Result at the same time is confusing and makes the code less readable.

Reply:
I've yet to read the code but I'll note that I usually find `Result<Option<...>, Error>` to be fine as long as the Option is indeed a successful case still. If `None` is meant to indicate an overall failure of some kind it should likely be part of the error enum instead.